packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -542,9 +545,30 @@ export class AzureCommunicationCallAdapter<AgentType extends CallAgent | BetaTea
   public async startCamera(options?: VideoStreamOptions): Promise<void> {
     return await this.asyncTeeErrorToEventEmitter(async () => {
       if (!isCameraOn(this.getState())) {
-        await this.handlers.onToggleCamera(options);
+        // First kick off the effect on the local device before starting the camera in the call.
+        // This prevents the effect not being applied for a brief moment when the camera is started.
         /* @conditional-compile-remove(video-background-effects) */
-        await startSelectedVideoEffect(this);
+        {
+          const selectedEffect = this.getState().selectedVideoBackgroundEffect;
+          const videoDeviceInfo = this.getState().devices.selectedCamera || this.getState().devices.cameras[0];
+          if (selectedEffect && videoDeviceInfo) {
+            const stream = new SDKLocalVideoStream(videoDeviceInfo);
```

Comment:
Do we need to create a new effect here? We have underlying handlers, shouldn't we call that. Another question about video device info, here the first camera is selected. Should the effect be applied to all video streams.

Reply:
Good questions!

> We have underlying handlers, shouldn't we call that

The handlers we have only act on active streams - as they should. Here we are applying an effect to a local video stream before turning it on. The only reason we can do this is we know we are turning that camera on right away.

> Could we update the handlers to always work on inactive cameras?

Yes, but then we would need extra logic to start and stop effects when the camera is switched. Here we don't need to worry becuase we are immediately turning on the camera. We could also move this logic into the onToggleCamera handler but that muddies its functionality and removes the reflective properties of the function mapping to the according react component.

> the first camera is selected. Should the effect be applied to all video streams

We shouldn't because we are applying the effect to the camera (i.e. the stream) we are immediately about to turn on (I'll update this bit code to use a common function that onStartCamera uses so they always use the same logic of grabbing the selected camera).